api/issuer.py

```diff
@@ -0,0 +1,114 @@
+        abort(die.status_code, die.public_msg)
+    current_app.logger.debug("... decoded token payload: {}".format(payload))
+
+    # Check TIL access depending on HTTP method
+    if request.method == 'POST':
+        # POST: Create issuer flow
+        
+        # Check for 'Create Issuer' role
+        try:
+            current_app.logger.debug("... checking for necessary role to create issuer")
+            if not check_create_role(payload, conf):
+                current_app.logger.debug("Required role was not found in JWT credential. Returning status 401.")
+                current_app.logger.debug("... required role '{}' for target DID '{}'".format(conf['issuer']['roles']['createRole'], conf['issuer']['providerId']))
+                abort(401, "Issued roles do not allow to create an issuer")
+        except IssuerException as cie:
+            current_app.logger.debug("Error when checking for required role: {}. Returning status {}.".format(cie.internal_msg, cie.status_code))
+            abort(cie.status_code, cie.public_msg)
+        
+    elif request.method == 'PUT':
+        # PUT: Update issuer flow
+
+        # Check for 'Update Issuer' role
+        try:
+            current_app.logger.debug("... checking for necessary role to update issuer")
+            if not check_update_role(payload, conf):
+                current_app.logger.debug("Required role was not found in JWT credential. Returning status 401.")
+                current_app.logger.debug("... required role '{}' for target DID '{}'".format(conf['issuer']['roles']['updateRole'], conf['issuer']['providerId']))
+                abort(401, "Issued roles do not allow to update an issuer")
+        except IssuerException as pie:
+            current_app.logger.debug("Error when checking for required role: {}. Returning status {}.".format(pie.internal_msg, pie.status_code))
+            abort(pie.status_code, pie.public_msg)
+    
+    elif request.method == 'DELETE':
+        # DELETE: Delete issuer flow
+
+        # Check for 'Delete Issuer' role
+        try:
+            current_app.logger.debug("... checking for necessary role to delete issuer")
+            if not check_delete_role(payload, conf):
+                current_app.logger.debug("Required role was not found in JWT credential. Returning status 401.")
+                current_app.logger.debug("... required role '{}' for target DID '{}'".format(conf['issuer']['roles']['deleteRole'], conf['issuer']['providerId']))
+                abort(401, "Issued roles do not allow to delete an issuer")
+        except IssuerException as delie:
+            current_app.logger.debug("Error when checking for required role: {}. Returning status {}.".format(delie.internal_msg, delie.status_code))
+            abort(delie.status_code, delie.public_msg)
+    
+    else:
+        # should not happen
+        abort(500, "Invalid HTTP method")
```

Comment:
If someone makes GET request to this endpoint, should he get 500 or 400 (since it's wrong request)?

Reply:
Flask won't accept it and will return some default error.
Therefore one shouldn't reach this else statement.

https://github.com/i4Trust/activation-service/blob/python-til/api/issuer.py#L14

Reply:
Oh, I see.